contrib/database/sql/conn.go

```diff
@@ -342,7 +366,7 @@ func (tp *traceParams) tryTrace(ctx context.Context, qtype QueryType, query stri
 			span.SetTag(k, v)
 		}
 	}
-	if err != nil && (tp.cfg.errCheck == nil || tp.cfg.errCheck(err)) {
+	if err != nil && !events.IsSecurityError(err) && (tp.cfg.errCheck == nil || tp.cfg.errCheck(err)) {
```

Comment:
If I understand this correctly, this mean users are still gonna see spans created for these functions but the actual db query is not executed, without any error or warning informing the query got blocked by appsec.

Is this the expected behaviour from the product side? Wouldn't it better to tag these spans with an error to be explicit that this happened? Users have the `WithErrorCheck` option to skip tagging this error if they find it too noisy.

Reply:
Indeed, users will see those spans, even though appsec will block the query execution.
This will be reflected in the tags of the root span, as well as visible in the security pannel of datadog, the same way it's done for http/grpc requests blocking.